mergin/client.py

```diff
@@ -57,21 +56,40 @@ def __init__(self, url=None, auth_token=None, login=None, password=None, plugin_
                 "username": token_data["username"]
             }
 
+        # Setup proxy handler, if needed
+        proxy_handler = None
+        auth_handler = None
+        if proxy_config is not None:
+            proxy_url_parsed = urllib.parse.urlparse(proxy_config['url'])
+            proxy_url = proxy_url_parsed.path
+            if proxy_config["user"] and proxy_config["password"]:
+                proxy_handler = urllib.request.ProxyHandler({
+                    "https": f"{proxy_config['user']}:{proxy_config['password']}@{proxy_url}:{proxy_config['port']}"
+                })
+                auth_handler = urllib.request.HTTPBasicAuthHandler()
+            else:
+                proxy_handler = urllib.request.ProxyHandler({"https": f"{proxy_url}:{proxy_config['port']}"})
+        handlers = []
+        if proxy_handler is not None:
+            handlers.append(proxy_handler)
+        if auth_handler is not None:
+            handlers.append(auth_handler)
```

Comment:
why not just add those proxy/auth handlers directly to `handlers` list when we create them? that would make the code easier to read and shorter :-)

Reply:
Done.